### PR TITLE
End reason implementation for Contracts

### PR DESCRIPTION
**Pull request by JoaFS**
This pull request have the implementation of the end reasons of contracts.
I took the existing "Role" implementation as a reference to implement this new function.

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/models/Contract.java [hunk 1]
```diff
@@ -168,4 +173,15 @@ public Integer getLatestPayment(Currency currency) {
...
     }
+
+    public EndReason getEndReason() {  return endReason; }
+
+    @Nullable
+    public UUID getEndReasonId() {
+        return endReason != null ? endReason.getId() : null;
+    }
+
+    public void setEndReason(EndReason endReason) { this.endReason = endReason; }
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Contract.java:184]: please fix format

**Review by eazaretzky — CHANGES_REQUESTED** [review]
only a few minor things to fix, good job!

File: src/main/java/com/entropyteam/entropay/employees/dtos/EndReasonDto.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.UUID;
+
+import com.entropyteam.entropay.employees.models.EndReason;
+
+import jakarta.validation.constraints.NotNull;
+
+public record EndReasonDto(UUID id,
+                           @NotNull(message = "Name is mandatory")
+                           String name) {
+    public EndReasonDto(EndReason endReason)    { this(endReason.getId(), endReason.getName());    }
```
[the remark this answers is not in the document]
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/dtos/EndReasonDto.java]: Fixed

File: src/main/java/com/entropyteam/entropay/employees/models/Contract.java [hunk 3]
```diff
@@ -168,4 +173,15 @@ public Integer getLatestPayment(Currency currency) {
                 .map(payment -> payment.getSalary().intValue())
                 .orElse(0);
     }
+
+    public EndReason getEndReason() {  return endReason; }
```
[the remark this answers is not in the document]
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Contract.java]: Fixed

File: src/main/java/com/entropyteam/entropay/employees/models/Contract.java (hunk 1 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Contract.java]: Fixed

File: src/main/java/com/entropyteam/entropay/employees/services/ContractService.java [hunk 4]
```diff
@@ -142,6 +150,7 @@ protected Contract toEntity(ContractDto entity) {
         contract.setSeniority(seniority);
         contract.setPaymentsSettlement(entity.paymentSettlement() == null ? Collections.emptySet() :
                 entity.paymentSettlement().stream().map(PaymentSettlement::new).collect(Collectors.toSet()));
+        contract.setEndReason(endReason);
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/services/ContractService.java:153]: @JoaFS another posible implementation of the same is this:
```
Optional.ofNullable(entity.endReasonId())
                .flatMap(endReasonRepository::findById)
                .ifPresent(contract::setEndReason);
```

Both are correct, I'm taking use of this opportunity just to show you some functionalities of the language

**Review by eazaretzky — APPROVED** [review]

**Pull request merged**